src/cephadm/cephadmlib/node_proxy/util.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import yaml
+import os
+import time
+import re
+import ssl
+from urllib.error import HTTPError, URLError
+from urllib.request import urlopen, Request
+from typing import Dict, List, Callable, Any, Optional, MutableMapping, Tuple
+
+
+class Logger:
+    _Logger: List['Logger'] = []
+
+    def __init__(self, name: str, level: int = logging.INFO):
+        self.name = name
+        self.level = level
+
+        Logger._Logger.append(self)
+        self.logger = self.get_logger()
+
+    def get_logger(self) -> logging.Logger:
+        logger = logging.getLogger(self.name)
+        logger.setLevel(self.level)
+        handler = logging.StreamHandler()
+        handler.setLevel(self.level)
+        fmt = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+        handler.setFormatter(fmt)
+        logger.handlers.clear()
+        logger.addHandler(handler)
+        logger.propagate = False
+
+        return logger
+
+
+class Config:
+
+    def __init__(self,
+                 config_file: str = '/etc/ceph/node-proxy.yaml',
+                 default_config: Dict[str, Any] = {}) -> None:
+        self.config_file = config_file
+        self.default_config = default_config
+
+        self.load_config()
+
+    def load_config(self) -> None:
+        if os.path.exists(self.config_file):
+            with open(self.config_file, 'r') as f:
+                self.config = yaml.safe_load(f)
+        else:
+            self.config = self.default_config
+
+        for k, v in self.default_config.items():
+            if k not in self.config.keys():
+                self.config[k] = v
+
+        for k, v in self.config.items():
+            setattr(self, k, v)
+
+        # TODO: need to be improved
+        for _l in Logger._Logger:
+            _l.logger.setLevel(self.logging['level'])  # type: ignore
+            _l.logger.handlers[0].setLevel(self.logging['level'])  # type: ignore
+
+    def reload(self, config_file: str = '') -> None:
+        if config_file != '':
+            self.config_file = config_file
+        self.load_config()
+
+
+log = Logger(__name__)
+
+
+def to_snake_case(name: str) -> str:
+    name = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
+    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', name).lower()
```

Comment:
is the need for snake case a redfish thing? Like some of the URLs you need to access are using snake case?

Reply:
it's more about the results (json) returned by redfish.
node-proxy was intended to be an interface between ceph-mgr and whatever the source of information would be which is RedFish at the moment but the code was thought so you could eventually 'plug' another type of source.
node-proxy must expose its data in a standardized way.